frontend/src/component/AddSpecies.tsx

```diff
@@ -0,0 +1,68 @@
+import {Habitat, NewSpecies} from "../model/Species";
+import {ChangeEvent, useState} from "react";
+
+type AddSpeciesProps = {
+    addSpecies: (newSpecies: NewSpecies) => Promise<any>;
+}
+
+export default function AddSpecies(props: AddSpeciesProps) {
+
+    const [name, setName] = useState<string>("");
+    const [waterDemand, setWaterDemand] = useState<number>(5);
+    const [fertilizerDemand, setFertilizerDemand] = useState<number>(5);
+    const [habitat, setHabitat] = useState<number>(1);
+    const [soil, setSoil] = useState<number>(1);
+
+    function onNameChange(event: ChangeEvent<HTMLInputElement>) {
+        setName(event.target.value)
+    }
+    function onWaterDemandChange(event: ChangeEvent<HTMLInputElement>){
+        setWaterDemand(parseInt(event.target.value))
+    }
+    function onFertilizerDemandChange(event: ChangeEvent<HTMLInputElement>){
+        setFertilizerDemand(parseInt(event.target.value))
+    }
+    function onHabitatChange(event: ChangeEvent<HTMLSelectElement>){
+        setHabitat(parseInt(event.target.value))
+    }
+    function onSoilChange(event: ChangeEvent<HTMLSelectElement>){
+        setSoil(parseInt(event.target.value))
+    }
+    function onClickSave(){
+        props.addSpecies({name: name,waterDemand: waterDemand,fertilizerDemand: fertilizerDemand,habitat: Habitat.Semishaded,soil:soil})
+            .then(() => {
+                setName("");
+                setWaterDemand(5);
+                setFertilizerDemand(5);
+                setHabitat(0);
+                setSoil(0)
+            })
+        console.log(props.addSpecies)
+    }
+
+    return (
+        <div>
+            <input type={"text"} placeholder={"Name"} value={name} onChange={onNameChange}/><p/>
+            <label htmlFor={"waterDemand"}>Water demand</label>
+            <input id={"waterDemand"} type={"range"} min={0} max={10} step={1} defaultValue={5} onChange={onWaterDemandChange}/><p/>
```

Comment:
So ein kleines Enter nach jedem Label würde der Lesbarkeit vlt gut tun 
Kleiner Tipp: Man kann das Input im Label nesten und nochmehr Ordnung schaffen

```js
            <label htmlFor={"waterDemand"}>Water demand
                <input id={"waterDemand"} type={"range"} min={0} max={10} step={1} defaultValue={5}
                       onChange={onWaterDemandChange}/><p/>
            </label>

Reply:
"A simple blank line does wonders for readability"